apis/run/v1alpha3/clusterbootstrap_types.go

```diff
@@ -4,12 +4,33 @@
 package v1alpha3
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ClusterBootstrapStatus defines the observed state of ClusterBootstrap
 type ClusterBootstrapStatus struct {
 	ResolvedTKR string `json:"resolvedTKR,omitempty"`
+
+	Conditions Conditions `json:"conditions,omitempty"`
+}
+
+type Conditions []Condition
+type ConditionType string
+
+type Condition struct {
+	// Type of condition in CamelCase
+	Type ConditionType `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown
+	Status corev1.ConditionStatus `json:"status"`
+
+	// A human readable error message. Will be empty in case no error happened
+	UsefulErrorMessage string `json:"usefulErrorMessage,omitempty"`
```

Comment:
Does ```Reason``` have to be machine readable and short? I ask cause this ```UsefulErrorMessage``` I have used here is longer and has is used for a descriptive error message

Reply:
So here is Carvel's ```Status``` and its nested ```appCondition``` types that I use to fill in for our purpose's. I only use top level ```UsefulErrorMessage``` and the ```Status``` and ```Type``` from conditions plus timestamp of change. I'm afraid if I use ```Message``` instead of ```UsefulErrorMessage```,  I mislead someone familiar with pkgi status fields also ```UsefulErrorMessage``` is not a short, machine understandable string.

```status.Conditions```:
```
type GenericStatus struct {
	// +optional
	ObservedGeneration int64
	// +optional
	Conditions []AppCondition
	// +optional
	FriendlyDescription string
	// +optional
	UsefulErrorMessage string
}

````

```
type AppCondition struct {
	Type   AppConditionType
	Status corev1.ConditionStatus
	// Unique, this should be a short, machine understandable string that gives the reason
	// for condition's last transition. If it reports "ResizeStarted" that means the underlying
	// persistent volume is being resized.
	// +optional
	Reason string
	// Human-readable message indicating details about last transition.
	// +optional
	Message string
}
```

Reply:
Ill defer this to you and @vijaykatam, but IMO we should not be concerned about people familiar with pkgi status fields (which seems an exceptions to me), but instead with people familiar with the rest of Kubernetes & Cluster API

Reply:
@fabriziopandini Is there any limitation or recommendation for the size and format of the ```Message``` field? This ```UsefulErrorMessage``` is human understandable error message and can be comparably long. Would this align with CAPI's api definition for the ```Message``` field?

Reply:
Changed to use CAPI's conditions and now I populate Message with UsefulErrorMessage